### ISSUE TITLE
Wrap JSON data in a CDATA element

### ISSUE THREAD

**Issue by billabel** [DannyBen/victor#67]
I'm trying to insert JSON data wrapped in a CDATA tag. Is there a way to do this? Or do I need to implement something like how you render CSS styles?

I've got the following, but the brackets get replaced with HTML entities, regardless if I escape them or not.

```
svg = Victor::SVG.new template: "app/assets/images/badges/template.svg"
svg.element "openbadges:assertion", verify: "url_here" do
  svg._ "<![CDATA[{ key: 'data' }]]>"
end
svg.render glue: ""
```

I'm trying to get to this output:

```
<openbadges:assertion verify="url_here">
<![CDATA[{ key: 'data' }]]>
</openbadges:assertion>
```

**Comment by billabel** [comment]
I came up with this fix, but it would be nice to have an option for element to wrap with CDATA.

```
file = svg.render glue: ""
file.gsub(/&lt;/, "<").gsub(/&gt;/, ">")
```

**Comment by DannyBen** [comment]
Well - while your solution of course works, I feel that if the use case is common, then Victor should provide you with a way to do it.

What is this used for? And how would you expect to be able to add such data to the SVG?
Maybe we can come up with a user friendly way of adding such data.

**Comment by DannyBen** [comment]
Also - I don't like this line at all:

```ruby
svg._ "<![CDATA[{ key: 'data' }]]>"
```

better to use `svg.element '_', ...`

**Comment by DannyBen** [comment]
Oh - I totally forgot.

1. You can use an exclamation mark suffix on the element's name to disable HTML encoding
2. You can provide a second argument to any element, and it will be just printed inside the element as is.

Any of these will work, the first option is the best/

```ruby
# Option 1
cdata = "<![CDATA[{ key: 'data' }]]>"
svg.element 'openbadges:assertion!', cdata, verify: 'url_here'

# Option 2, with a block
svg.element 'openbadges:assertion', verify: 'url_here' do
  svg.element '_!', "<![CDATA[{ key: 'data' }]]>"
end

# Option 3, with build
svg.build do
  element 'openbadges:assertion', verify: 'url_here' do
    element '_!', "<![CDATA[{ key: 'data' }]]>"
  end
end
```

**Comment by billabel** [comment]
Thank you. That's excellent. I think Option 1 is what I was looking for... I overlooked the option to disable XML encoding.

We are implementing the OpenBadge standard for digital credentials. The standard uses a 'baked' SVG which has JSON data inserted as shown above. https://openbadges.org/

I installed Victor and played around with it last night and this morning. It works great. Thanks for making it.
